### PR TITLE
NFT Refactor: remove redux

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
Previous state, as of [last PR](https://github.com/rainbow-me/rainbow/pull/4728):
* Selected wallet gets nfts from redux, external wallets get nfts from `useLegacyNFTs` hook

Current state:
* All wallets get nfts from `useLegacyNFTs` hook

### CODE REVIEW

File: src/handlers/localstorage/accountLocal.ts [hunk 1]
```diff
@@ -266,40 +266,6 @@ export const saveLocalENSRegistrations = (
     network
   );
 
-/**
```
Comment (benisgold) [inline src/handlers/localstorage/accountLocal.ts:269]: async storage for nfts is now handled by react query

File: src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx [hunk 2]
```diff
@@ -67,27 +51,6 @@ export default function NFTBriefTokenInfoRow({
 
   const currentPrice = asset?.currentPrice ?? listingValue;
 
-  useEffect(() => {
```
Comment (benisgold) [inline src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx:70]: all nfts now come from simplehash which includes floor price, so we no longer need this opensea fallback

File: src/hooks/useRefreshAccountData.ts [hunk 3]
```diff
@@ -27,19 +28,21 @@ export default function useRefreshAccountData() {
       return Promise.all([delay(1250)]);
     }
 
+    queryClient.invalidateQueries({
```
Comment (benisgold) [inline src/hooks/useRefreshAccountData.ts:31]: this invalidates the current nft cache which forces a refetch

File: src/redux/data.ts [hunk 4]
```diff
@@ -733,7 +734,9 @@ export const transactionsReceived = (
 
   if (appended && potentialNftTransaction) {
     setTimeout(() => {
-      dispatch(uniqueTokensRefreshState());
+      queryClient.invalidateQueries({
```
Comment (benisgold) [inline src/redux/data.ts:737]: this invalidates the current nft cache which forces a refetch

File: src/hooks/useLoadAccountData.ts [hunk 5]
```diff
@@ -23,17 +22,16 @@ export default function useLoadAccountData() {
       // tokens + nfts
       if (network === networkTypes.mainnet) {
         const p1 = dispatch(dataLoadState());
-        const p2 = dispatch(uniqueTokensLoadState());
```
Comment (benisgold) [inline src/hooks/useLoadAccountData.ts:26]: no longer need to prefetch nfts bc unlike redux, react query will fetch the nfts from the interweb (or cache if available) on demand

File: src/hooks/useInitializeAccountData.ts [hunk 6]
```diff
@@ -17,11 +16,6 @@ export default function useInitializeAccountData() {
         dispatch(explorerInit());
       });
 
-      InteractionManager.runAfterInteractions(async () => {
-        logger.sentry('Initialize uniqueTokens');
-        await dispatch(uniqueTokensRefreshState());
```
Comment (benisgold) [inline src/hooks/useInitializeAccountData.ts:22]: no longer need to prefetch nfts bc unlike redux, react query will fetch the nfts from the interweb (or cache if available) on demand

File: src/hooks/useLoadAccountData.ts (hunk 5 above)
Comment (benisgold) [inline src/hooks/useLoadAccountData.ts:26]: this redux action got nfts from local storage and put them into redux. this happens automatically in react query

File: src/hooks/useCollectible.ts [hunk 7]
```diff
@@ -46,25 +44,38 @@ export default function useCollectible(
 function useRevalidateInBackground({
```
[the remark this answers is not in the document]
Comment (estrattonbailey) [reply src/hooks/useCollectible.ts:44]: Ok I see removed in follow-up PR, sounds good

**Pull request merged**